specs/online_order_spec.rb

```diff
@@ -3,50 +3,108 @@
+ require_relative '../lib/customer'
 # You may also need to require other classes here
 
 # Because an OnlineOrder is a kind of Order, and we've
 # already tested a bunch of functionality on Order,
 # we effectively get all that testing for free! Here we'll
 # only test things that are different.
 
-xdescribe "OnlineOrder" do
+describe "OnlineOrder" do
   describe "#initialize" do
     it "Is a kind of Order" do
-      # Check that an OnlineOrder is in fact a kind of Order
-
       # Instatiate your OnlineOrder here
-      # online_order =
-      # online_order.must_be_kind_of Grocery::Order
+      online_order = Grocery::OnlineOrder.new(8, "b", 25, "d")
+      #puts "online_order: #{online_order}"
+
+      ##works checks initialize
+      # online_order2 = Grocery::OnlineOrder.new("12", "b", "35")
+      #   puts "online_order2: #{online_order2}"
+      #   array = []
+      #   array << online_order2
+      #   array.each do |check|
+      #     puts check.id.class
+      #     puts check.products.class
+      #       cust_id_array = []
+      #       cust_id_array << check.customer_id
+      #         cust_id_array.each do |check|
+      #           puts check.id.class
+      #           puts check.email
+      #           puts check.address
+      #         end
+      #     puts check.status
+      #   end
+
+      # Check that an OnlineOrder is in fact a kind of Order
+      online_order.must_be_kind_of Grocery::Order
     end
 
     it "Can access Customer object" do
       # TODO: Your test code here!
+
+      online_order2 = Grocery::OnlineOrder.new(12, "b", 35, "g")
+
+      # PRINTS
+      #   puts "online_order2: #{online_order2}"
+      #   array = []
+      #   array << online_order
+      #   array.each do |check|
+      #     puts "customer id: #{check.customer_id}"
+      #       cust_id_array = []
+      #       cust_id_array << check.customer_id
+      #         cust_id_array.each do |check|
+      #           puts check.id
+      #           puts check.email
+      #           puts check.address
+      #         end
+      #   end
+
     end
 
     it "Can access the online order status" do
       # TODO: Your test code here!
+      online_order3 = Grocery::OnlineOrder.new(8, "b", 5, "shipped")
+      #puts online_order3
+      array = []
+      array << online_order3
+        array.each do |check|
+          print check.status == :shipped
+        end
     end
   end
 
   describe "#total" do
     it "Adds a shipping fee" do
       # TODO: Your test code here!
+      order = Grocery::OnlineOrder.new(7, {Eggs:84.23,Watermelon:11.16,Cherries:10.4}, 24, :complete)
+      order_plus_shipping = order.total
+      puts "Total_3_Products: #{order_plus_shipping}"
+
     end
 
     it "Doesn't add a shipping fee if there are no products" do
       # TODO: Your test code here!
+      order = Grocery::OnlineOrder.new(7, {}, 24, :complete)
+      order_plus_shipping = order.total
+      puts "Total_0_Products: #{order_plus_shipping}"
```

Comment:
Don't forget to remove all `puts` statements before turning in. `puts` does not create a valid test.